esp-hal/src/spi/master.rs

```diff
@@ -526,6 +512,11 @@ where
         Ok(())
     }
 
+    /// Read bytes from SPI.
+    pub fn read_bytes(&mut self, words: &mut [u8]) -> Result<(), Error> {
```

Comment:
We'll have to take a bit of care here. UART's `read_bytes` is not blocking, SPI's `read_bytes` is blocking. As the peripherals work differently, it makes logical sense, but I'm not sure if we want the function use the same name, but do different things.

Reply:
I guess this should be resolved in https://github.com/esp-rs/esp-hal/pull/2882 which is a bit stalled due to this PR

Reply:
Imo this is fine as is, and if anything we should change uart's `read_bytes` back to being blocking and introduce a nb API post 1.0

Reply:
A little bit more documentation wouldn't hurt probably. "Read bytes from SPI" is very brief

Reply:
I'll do it once these PRs land.

Reply:
Please leave a non-blocking one around, in my application I don't know how much data the transmitter is going to send so I need the non-blocking version to just read what is available 😅